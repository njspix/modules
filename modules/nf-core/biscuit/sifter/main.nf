process SIFTER {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-3f58ec2bbcefe75f55f36cf5defe6d77c8803de5:e34531e20dd2ff3fb3a8b1aa9ab20a205bbdd7e1-0':
        'biocontainers/mulled-v2-3f58ec2bbcefe75f55f36cf5defe6d77c8803de5:e34531e20dd2ff3fb3a8b1aa9ab20a205bbdd7e1-0' }"

    input:
    tuple val(meta), path(reads)
    path index

    output:
    tuple val(meta), path("*.bam")                , emit: bam
    tuple val(meta), path("*.bai")                , emit: bai
    tuple val(meta), path("*_dupsifter_stats.txt"), emit: dupsifter_stats
    path "versions.yml"                           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def args = task.ext.args ?: ''
    def args2 = task.ext.args2 ?: ''
    def args3 = task.ext.args3 ?: ''
    def biscuit_cpus = (int) Math.max(Math.floor(task.cpus*0.95),1)
    def samtools_cpus = Math.max(task.cpus-biscuit_cpus, 1)
    """
    INDEX=`find -L ./ -name "*.bis.amb" | sed 's/\\.bis.amb\$//'`

    biscuit align \\
        -@ $biscuit_cpus \\
        $args \\
        \$INDEX \\
        $reads | \\
    dupsifter \\
        $args2 \\
        --stats-output ${prefix}_dupsifter_stats.txt \\
        \$INDEX | \\
    samtools sort \\
        -@ $samtools_cpus \\
        $args3 \\
        --write-index \\
        -o ${prefix}.bam##idx##${prefix}.bam.bai

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        biscuit: \$( biscuit version |& sed '1!d; s/^.*BISCUIT Version: //' )
        dupsifter: \$( dupsifter --version |& sed '2!d; s/Version: //' )
        samtools: \$( samtools --version |& sed '1!d; s/^.*samtools //' )
    END_VERSIONS
    """
}
