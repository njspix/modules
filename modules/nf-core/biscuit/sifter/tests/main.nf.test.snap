{
    "sarscov2 - [bam, bai] - BiscuitIndex | paired end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam:md5,c4812f543fec813dfd370332d589d473"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam.bai:md5,8ed9a1c24274989fdc2ac6909f57bfed"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_dupsifter_stats.txt:md5,9d8350d1d685600426befabe91c26744"
                    ]
                ],
                "3": [
                    "versions.yml:md5,f3951bcc0dcdd34cf5c56b726c371b59"
                ],
                "bai": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam.bai:md5,8ed9a1c24274989fdc2ac6909f57bfed"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam:md5,c4812f543fec813dfd370332d589d473"
                    ]
                ],
                "dupsifter_stats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_dupsifter_stats.txt:md5,9d8350d1d685600426befabe91c26744"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f3951bcc0dcdd34cf5c56b726c371b59"
                ]
            }
        ],
        "timestamp": "2024-01-23T17:44:02.476974072"
    },
    "sarscov2 - [bam, bai] - BiscuitIndex | single end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam:md5,988717cf701473f6283b8035c202300c"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam.bai:md5,fd49e419998d4b79997370b9ca2bbfb3"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_dupsifter_stats.txt:md5,b953677ce0118555fd1c1b679584489d"
                    ]
                ],
                "3": [
                    "versions.yml:md5,f3951bcc0dcdd34cf5c56b726c371b59"
                ],
                "bai": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam.bai:md5,fd49e419998d4b79997370b9ca2bbfb3"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam:md5,988717cf701473f6283b8035c202300c"
                    ]
                ],
                "dupsifter_stats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_dupsifter_stats.txt:md5,b953677ce0118555fd1c1b679584489d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f3951bcc0dcdd34cf5c56b726c371b59"
                ]
            }
        ],
        "timestamp": "2024-01-23T17:43:46.772738816"
    }
}